hwilib/devices/trezor.py

```diff
@@ -258,6 +258,26 @@ def ignore_input():
                     else:
                         raise BadArgumentError("Output is not an address")
 
+                # Add the derivation path for change, but only if there is exactly one derivation path
+                psbt_out = tx.outputs[i]
+                if len(psbt_out.hd_keypaths) == 1:
+                    _, keypath = next(iter(psbt_out.hd_keypaths.items()))
+                    if keypath[0] == master_fp:
+                        wit, ver, prog = out.is_witness()
+                        if out.is_p2pkh():
+                            txoutput.address_n = keypath[1:]
+                            txoutput.address = None
+                        elif wit:
+                            txoutput.script_type = proto.OutputScriptType.PAYTOWITNESS
+                            txoutput.address_n = keypath[1:]
+                            txoutput.address = None
```

Comment:
without being able to test this locally, does this mean that *any* output with known derivation path will be considered change and elided from the screen?

I do a slightly hackier check for ledger, checking that 2nd to last index is 1(hdsplit): https://github.com/bitcoin-core/HWI/blob/master/hwilib/devices/ledger.py#L158 but in my real-world use it behaves much better for self-sending scenarios.

Reply:
> without being able to test this locally, does this mean that _any_ output with known derivation path will be considered change and elided from the screen?

It would appear it does. If the transaction is fully send to self, then no amounts are shown on screen. It does still check that the address derived matches the address given and if it does not match, it will be shown.

Reply:
:man_shrugging: ok. Maybe document that, keeping logic the same.

Reply:
Added a note in `docs/trezor.md`